sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -586,9 +608,21 @@ object ContainsExpressionBuilder extends StringBinaryPredicateExpressionBuilderB
 }
 
 case class Contains(left: Expression, right: Expression) extends StringPredicate {
-  override def compare(l: UTF8String, r: UTF8String): Boolean = l.contains(r)
+  override def compare(l: UTF8String, r: UTF8String): Boolean = {
+    val collationId: Int = left.dataType.asInstanceOf[StringType].collationId
```

Comment:
Can you push collationId as class property, outside of of `compare`?

That way you can share it between `doGenCode` and `compare`  + it should be more efficient to do cast once then per row.

Reply:
makes sense, avoid duplicate eval